src/App.tsx

```diff
@@ -8,7 +8,82 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState('');
+  const [selectValue, setSelectValue] = useState('');
+  const [clickedValue, setClickedValue] = useState(0);
+  const [isModal, setIsModal] = useState(false);
+
+  useEffect(() => {
+    getTodos().then(setTodos);
+  }, []);
+
+  let copyTodo = todos;
+
+  const filterList = (search: string) => {
+    setInputValue(search.toLowerCase());
+  };
+
+  if (inputValue.length > 0) {
+    copyTodo = todos.filter((todo) => todo.title.toLowerCase()
+      .includes(inputValue
+        .toLowerCase()));
+  }
+
+  if (selectValue.length > 0) {
+    switch (selectValue) {
+      case 'all':
+        setInputValue('');
+        setSelectValue('');
+        break;
+      case 'completed':
+        if (inputValue.length === 0) {
+          copyTodo = todos.filter((todo) => todo.completed);
+        } else {
+          copyTodo = todos.filter(
+            (todo) => todo.completed
+              && todo.title.toLowerCase()
+                .includes(inputValue.toLowerCase()),
```

Comment:
maybe try moving it to separate function not to rewrite logic?
`todo.title.toLowerCase().includes(inputValue.toLowerCase())` is used twice